go/interpreter/lfvm/memory_test.go

```diff
@@ -185,3 +185,120 @@ func TestMemory_expandMemory_expandsMemoryOnlyWhenNeeded(t *testing.T) {
 		})
 	}
 }
+
+func TestMemory_SetByte_SuccessfulCases(t *testing.T) {
+
+	baseData := []byte{0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef}
+
+	tests := map[string]struct {
+		memory []byte
+		offset uint64
+	}{
+		"write to empty memory": {},
+		"write to first element of memory": {
+			memory: baseData,
+			offset: 0,
+		},
+		"write with offset within memory": {
+			memory: baseData,
+			offset: 4,
+		},
+		"offset trigger expansion of empty memory": {
+			offset: 2,
+		},
+		"offset trigger memory expansion": {
+			memory: baseData[:4],
+			offset: 8,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctxt := getEmptyContext()
+			m := NewMemory()
+			m.store = test.memory
```

Comment:
I commented on this before: when you assign the `test.memory` to `m.store` here and `test.memory` is an alias of `baseData`, the operation `m.setByte` below will write into `baseData`. Since all the tests are pointing to the same `baseData`, this makes them all inter-dependent.

If you want to check that this is true, print `baseData` before each test. It changes its value.

In this test case it is not a problem. But it would be if, for instance, you check that only this one byte got changed (which would be a nice addition).